pandas/plotting/_matplotlib/core.py

```diff
@@ -413,7 +413,13 @@ def _compute_plot_data(self):
         # np.ndarray before plot.
         numeric_data = numeric_data.copy()
         for col in numeric_data:
-            numeric_data[col] = np.asarray(numeric_data[col])
+
+            # GH32073: cast to float if values contain nulled integers
+            values = numeric_data[col]
+            if values.isna().any().all():
+                values = values.astype(float)
+
```

Comment:
It seems like date-like data can be included here, and we don't want to convert those to floats.

I think this should be restricted to

```python
if is_integer_dtype(values.dtype):
    values = values.to_numpy(dtype="float", na_value=np.nan)
```